### PR TITLE
support governance's integration test

### PR DESCRIPTION
**Pull request by JiekerTime**

### CODE REVIEW

**Comment by menghaoranss** [comment]
@JiekerTime Some errors occurred.

**Comment by JiekerTime** [comment]
> @JiekerTime Some errors occurred.

Ok, I have fixed it.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/ShardingSphereAdapterContainer.java [hunk 1]
```diff
@@ -72,6 +72,14 @@ private YamlUserConfiguration loadAuthentication(final ParameterizedArray parame
      */
     public abstract DataSource getDataSource();
 
+    /**
+     * Get Governance DataSource.
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/ShardingSphereAdapterContainer.java:76]: Get governance data source

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/ShardingSphereAdapterContainer.java [hunk 2]
```diff
@@ -72,6 +72,14 @@ private YamlUserConfiguration loadAuthentication(final ParameterizedArray parame
...
+     *
+     * @param serverLists server list
+     * @return DataSource
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/ShardingSphereAdapterContainer.java:79]: return data source

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereJDBCContainer.java [hunk 3]
```diff
@@ -66,14 +72,35 @@ public void start() {
      */
     public DataSource getDataSource() {
         try {
-            if ("sharding_governance".equals(getParameterizedArray().getScenario())) {
-                return YamlGovernanceShardingSphereDataSourceFactory.createDataSource(new File(EnvironmentPath.getRulesConfigurationFile(getParameterizedArray().getScenario())));
-            }
             return YamlShardingSphereDataSourceFactory.createDataSource(dataSourceMap, new File(EnvironmentPath.getRulesConfigurationFile(getParameterizedArray().getScenario())));
         } catch (SQLException | IOException ex) {
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereJDBCContainer.java:76]: final

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereJDBCContainer.java [hunk 4]
```diff
@@ -66,14 +72,35 @@ public void start() {
...
+            YamlGovernanceConfiguration governance = configurations.getGovernance();
+            governance.getRegistryCenter().setServerLists(serverLists);
+            Properties properties = configurations.getProps();
+            if (configurations.getRules().isEmpty() || dataSourceMap.isEmpty()) {
+                return new GovernanceShardingSphereDataSource(YamlGovernanceConfigurationSwapperUtil.marshal(governance));
+            } else {
+                return new GovernanceShardingSphereDataSource(dataSourceMap, new YamlRuleConfigurationSwapperEngine().swapToRuleConfigurations(configurations.getRules()), properties,
+                        YamlGovernanceConfigurationSwapperUtil.marshal(governance));
+            }
+        } catch (SQLException | IOException ex) {
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereJDBCContainer.java:100]: final

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereProxyContainer.java [hunk 5]
```diff
@@ -110,7 +110,17 @@ public DataSource getDataSource() {
         }
         return dataSourceProvider.get();
     }
-    
+
+    /**
+     * Get DataSource.
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereProxyContainer.java:115]: Get governance data source

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereProxyContainer.java [hunk 6]
```diff
@@ -110,7 +110,17 @@ public DataSource getDataSource() {
...
+     *
+     * @param serverLists server list
+     * @return DataSource
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/adapter/impl/ShardingSphereProxyContainer.java:118]: return data source

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/governance/ZookeeperContainer.java [hunk 7]
```diff
@@ -22,10 +22,19 @@
...
     public ZookeeperContainer(final ParameterizedArray parameterizedArray) {
         super("zookeeper", "zookeeper:3.6.2", false, parameterizedArray);
         setWaitStrategy(new LogMessageWaitStrategy().withRegEx(".*PrepRequestProcessor \\(sid:[0-9]+\\) started.*"));
     }
-    
+
+    /**
+     * Get server list.
+     *
+     * @return serverLists
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/governance/ZookeeperContainer.java:34]: return server list

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/governance/ZookeeperContainer.java [hunk 8]
```diff
@@ -22,10 +22,19 @@
 import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
 
 public class ZookeeperContainer extends ShardingSphereContainer {
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/governance/ZookeeperContainer.java:24]: Need final ?

**Review by menghaoranss — CHANGES_REQUESTED** [review]

**Review by menghaoranss — APPROVED** [review]

**Pull request merged**